lib/phoenix/controller.ex

```diff
@@ -110,6 +110,19 @@ defmodule Phoenix.Controller do
   def action_name(conn), do: conn.private.phoenix_action
 
   @doc """
+  Returns the controller name as a string, raises if unavailable.
+  """
+  @spec controller_name(Plug.Conn.t) :: String.t
+  def controller_name(conn) do
```

Comment:
The implementation of this function should be just:

    Phoenix.Naming.resource_name(conn.private.private_controller, "Controller")